chef/cookbooks/provisioner/templates/suse/crowbar_register.erb

```diff
@@ -283,6 +296,18 @@ curl -L -o /etc/chef/validation.pem \
     --connect-timeout 60 -s \
     "$HTTP_SERVER/validation.pem"
 
+nodes=$(get_nodes_allow_fail || echo "")
+if test -z "$nodes"; then
+  echo "Failed to list nodes known on the administration server."
+  echo "Is the administration server up?"
+  exit 1
+fi
+
+if echo "$nodes" | grep -q "\"name\":\"$HOSTNAME\""; then
```

Comment:
it feels simpler to me to just query for 

/crowbar/machines/1.0/$HOSTNAME

is also slightly faster (show instead of list I guess)